pkg/controllers/dynakube/activegate/internal/statefulset/reconciler.go

```diff
@@ -77,18 +78,34 @@ func (r *Reconciler) manageStatefulSet(ctx context.Context) error {
 	}
 
 	created, err := r.createStatefulSetIfNotExists(ctx, desiredSts)
-	if created || err != nil {
+	if err != nil {
+		conditions.SetKubeApiError(r.dynakube.Conditions(), ActiveGateStatefulSetConditionType, err)
+
 		return errors.WithStack(err)
+	} else if created {
+		conditions.SetStatefulSetCreated(r.dynakube.Conditions(), ActiveGateStatefulSetConditionType, desiredSts.Name)
+
+		return nil
 	}
 
 	deleted, err := r.deleteStatefulSetIfSelectorChanged(ctx, desiredSts)
-	if deleted || err != nil {
+	if err != nil {
+		conditions.SetKubeApiError(r.dynakube.Conditions(), ActiveGateStatefulSetConditionType, err)
+
 		return errors.WithStack(err)
+	} else if deleted {
+		conditions.SetStatefulSetDeleted(r.dynakube.Conditions(), ActiveGateStatefulSetConditionType, desiredSts.Name)
+
+		return nil
 	}
```

Comment:
is this condition needed?
also the `return nil` is probably incorrect

Reply:
That's at least the logic we [already have](https://github.com/Dynatrace/dynatrace-operator/blob/f3eba0fea8dfa9b10e9059b17e8d828ee55c47c2/pkg/controllers/dynakube/activegate/internal/statefulset/reconciler.go#L85).

Reply:
Nevertheless, removed it now

Reply:
Updated to call create afterwards instead of returning